### PR TITLE
PD-1440 session management

### PR DESCRIPTION
**Pull request by grvgoel81**
Adding session management support and changing UX designs.

### CODE REVIEW

File: core/build.gradle [hunk 1]
```diff
@@ -8,7 +8,7 @@ android {
     compileSdkVersion 31
 
     defaultConfig {
-        minSdkVersion 21
+        minSdkVersion 23
```
Comment (chaitanyapotti) [inline core/build.gradle:11]: shouldn't this be 24?
cause web3j min is 24

Comment (grvgoel81) [reply core/build.gradle:11]: core-sdk  internally has check for Android API level like in Web3Auth.kt like :
 if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
            initiateKeyStoreManager()
   }

So keyStore and web3J library will work only for and above Android API level 24. Anyways we can update sdk to support min API level 24.

File: core/src/main/java/com/web3auth/core/Web3Auth.kt [hunk 2]
```diff
@@ -102,23 +130,124 @@ class Web3Auth(web3AuthOptions: Web3AuthOptions) {
             logoutCompletableFuture.complete(null)
         }
 
+        web3AuthResponse.sessionId?.let { KeyStoreManagerUtils.encryptData(KeyStoreManagerUtils.SESSION_ID, it) }
```
Comment (chaitanyapotti) [inline core/src/main/java/com/web3auth/core/Web3Auth.kt:133]: what's the key it's being stored under?
it should be verifier + " | " + verifierId

Comment (grvgoel81) [reply core/src/main/java/com/web3auth/core/Web3Auth.kt]: Was using key "sessionId". But updated now.

**Review by chaitanyapotti — APPROVED** [review]
lgtm

**Pull request merged**